utils/config.go

```diff
@@ -689,3 +709,75 @@ func IsLeader() bool {
 		return true
 	}
 }
+
+func ValidateAmazonS3Endpoint(cfg *model.Config) {
+	_, ok := awsS3EndpointMap[*cfg.FileSettings.AmazonS3Endpoint]
+	if !ok {
+		*cfg.FileSettings.AmazonS3Endpoint = model.FILE_SETTINGS_DEFAULT_AMAZON_S3_ENDPOINT
+		l4g.Warn(T("utils.config.set_amazon_endpoint"), model.FILE_SETTINGS_DEFAULT_AMAZON_S3_ENDPOINT)
+	}
+}
+
+func ValidateAmazonS3Region(cfg *model.Config) {
+	valid := false
+	for _, region := range awsS3EndpointMap {
+		if region == *cfg.FileSettings.AmazonS3Region {
+			valid = true
+			break
+		}
+	}
+
+	if !valid {
+		*cfg.FileSettings.AmazonS3Region = model.FILE_SETTINGS_DEFAULT_AMAZON_S3_REGION
+		l4g.Warn(T("utils.config.set_amazon_region"), model.FILE_SETTINGS_DEFAULT_AMAZON_S3_REGION)
+	}
+}
```

Comment:
It's a little unintuitive that these validation functions aren't actually returning anything. Even if you're not using it, it would be nice to return at least a `bool` for valid or not

Reply:
I didn't think it was worth commenting, but I had exactly the same thought. In general, we should prefer pure functions rather than functions that operate only through side-effects.